### ISSUE TITLE
getResource selector override when using denormalizer

### ISSUE THREAD

**Issue by dpellier** [Brigad/redux-rest-easy#72]
**Context**

On one screen mount, I need to load a profile and its type.

For that I have the following endpoints (which I cannot change):
- /profile
- /profile/type

(this is dumb data for the issue but the concept is the same)

**Resources**
So I've created to resources for those:
```
const profileResource = createResource(profileKey)({
    fetch: {
        method: 'GET',
        url: `${API_URL}/profile`,
        normalizer: (profile) => normalize(profile, profileSchema)
    }
});
const {
    actions: {
        fetch: { perform: fetchProfile }
    },
    selectors: {
        resource: { getResource: getAllProfile }
    }
} = profileResource;
```
```
const profileTypeResource = createResource(profileTypeKey)({
    fetch: {
        method: 'GET',
        url: `${API_URL}/profile/type`,
        normalizer: (profileType) => normalize(profileType, profileTypeSchema)
    }
});
const {
    actions: {
        fetch: { perform: fetchProfileType }
    },
    selectors: {
        resource: { getResource: getAllProfileType }
    }
} = profileTypeResource;
```

And in my view, I fetch both:
```
class Profile extends Component {
    componentDidMount() {
        this.props.fetchProfile();
        this.props.fetchProfileType();
    }

    render() {
        ....
    }
}
function mapStateToProps(state) {
    return {
        profile: getAllProfile(state)[0],
        type: getAllProfileType(state)[0]
    };
}

function mapDispatchToProps (dispatch) {
    return {
        fetchProfile: (id) => dispatch(fetchProfile({ urlParams: { id } })),
        fetchProfileType: (id) => dispatch(fetchProfileType({ urlParams: { id } })),
    };
}
```

As I don't have any resources ID due to the API implementation, I use the `getResource` selector instead of the `getResourceById`.

So far, everything works fine.

**Issue**
Then I want to cast each of those resources into a specific class. For that I use the `denormalizer`.

I add the first one on Profile:
```
const profileResource = createResource(profileKey, {
    denormalizer: (resourceIds, resources) => resourceIds.map((id) => new Profile(resources[profileKey][id])) }
})({
    fetch: {
        method: 'GET',
        url: `${API_URL}/profile`,
        normalizer: (profile) => normalize(profile, profileSchema)
    }
});
```

If I check on the view at this point I get the correct objects:
```
render() {
     this.props.profile => undefined > Profile > Profile
     this.props.type => undefined > undefined > Object
}
```

Then I add the same to the second resource:
```
const profileTypeResource = createResource(profileTypeKey, {
    denormalizer: (resourceIds, resources) => resourceIds.map((id) => new ProfileType(resources[profileTypeKey][id])) }
})({
    fetch: {
        method: 'GET',
        url: `${API_URL}/profile/type`,
        normalizer: (profileType) => normalize(profileType, profileTypeSchema)
    }
});
```

And now I get:
```
render() {
     this.props.profile => undefined > Profile > ProfileType
     this.props.type => undefined > undefined > ProfileType
}
```

I thought initially that I was casting all resources again on denormalizer, but when logging, I see that each denormalizer only pass on its own resources.

I've also tried to switch one resources to `getResourceById` (by faking the id) and it goes well again.

**TL;DR**
Calling two `getResource` in `mapStateToProps` when each have a `denormalizer` applied overrides the resources and return the last one for each.

**Comment by adrienharnay** [comment]
Hey, thanks for opening an issue!

I'm sensing it will be hard to reproduce 😅 Could you first try replacing both your selectors

```
    selectors: {
        resource: { getResource: getAllProfile }
    }

    selectors: {
        resource: { getResource: getAllProfileType }
    }
```

with

```
    selectors: {
        fetch: { getResource: getAllProfile }
    }

    selectors: {
        fetch: { getResource: getAllProfileType }
    }
```

and your mapStateToProps

```
function mapStateToProps(state) {
    return {
        profile: getAllProfile(state)[0],
        type: getAllProfileType(state)[0]
    };
}
```

with

```
function mapStateToProps(state, ownProps) {
    return {
        profile: getAllProfile(state, ownProps)[0],
        type: getAllProfileType(state, ownProps)[0]
    };
}
```
?

Based on the result, I should have a better understanding of the problem 🙂 Thanks!

**Comment by adrienharnay** [comment]
Closing for now, feel free to reopen if you are still facing this issue 🙂
